From e9d4a632cba137d37f1c2eae985613503608f765 Mon Sep 17 00:00:00 2001
From: Daniel Stone <daniels@collabora.com>
Date: Mon, 30 Nov 2015 16:07:19 +0000
Subject: [PATCH 12/94] compositor-drm: Use fb->fd consistently
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Everyone else uses fb->fd rather than pulling the FD back out of GBM.
Use that in the destroy callback too.

Signed-off-by: Daniel Stone <daniels@collabora.com>
Reviewed-by: Armin KrezoviÄ‡ <krezovic.armin@gmail.com>

Differential Revision: https://phabricator.freedesktop.org/D1406
---
 src/compositor-drm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index b3c5f33..b24e1b3 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -249,10 +249,9 @@ static void
 drm_fb_destroy_callback(struct gbm_bo *bo, void *data)
 {
 	struct drm_fb *fb = data;
-	struct gbm_device *gbm = gbm_bo_get_device(bo);
 
 	if (fb->fb_id)
-		drmModeRmFB(gbm_device_get_fd(gbm), fb->fb_id);
+		drmModeRmFB(fb->fd, fb->fb_id);
 
 	weston_buffer_reference(&fb->buffer_ref, NULL);
 
-- 
1.9.1

